### ISSUE TITLE
flambda seems to increase a lot the overhead of loading native libraries

### ISSUE THREAD

**Issue by Blaisorblade** [coq/coq#13850]
#### Description of the problem

Split out of #11107.

That issue suggests the overhead of #13849 is significantly bigger when ocaml is compiled with flambda on, but @Zimmi48 points out that was always mysterious. flambda-optimized libraries might be slower to load, but the size of the slowdown might seem surprising:

> `time make -j1` for stdpp master. Coq master, ocaml 4.09.0, macOS 10.13.

||  | ¬ native-compiler
-|-|-
vanilla | 1m48s | 1m23s
flambda | 2m21s | 1m23s (the same)

Ideally, this should be investigated more carefully; however, this appears low priority at the moment.

Beware: IIRC native-compute (used to?) always enable flambda. However, it appears that `vanilla` vs `flambda` describes the underlying ocaml version*, explaining the difference.

*At least as of https://github.com/Blaisorblade/my-ocaml-coq-benchs/commit/b70dcf3cb8aff23ac167225155f8921dab764899.

**Comment by Blaisorblade** [comment]
cc @silene as well.

**Comment by ppedrot** [comment]
What was the exact setting for this slowdown? This was with native compilation set to `ondemand` and the stdlib native files available, right?

**Comment by Blaisorblade** [comment]
Yep, I tried to clarify the table of _old_ data — but I don't want to distract from other things like #13849.
